packages/task/src/node/task-server.ts

```diff
@@ -112,6 +126,34 @@ export class TaskServerImpl implements TaskServer, Disposable {
                             problems
                         });
                     }
+                    if (taskConfiguration.isBackground) {
+                        const backgroundTaskStatus = this.backgroundTaskStatusMap.get(event.taskId)!;
+                        if (!backgroundTaskStatus.isActive) {
+                            // Get the 'activeOnStart' value of the problem matcher 'background' property
+                            const activeOnStart = collector.problemMatchers[0]!.watching!.activeOnStart;
+                            if (activeOnStart) {
+                                backgroundTaskStatus.isActive = true;
+                            } else {
+                                const isBeginsPatternMatch = collector.matchBeginMatcher(event.line);
+                                if (isBeginsPatternMatch) {
+                                    backgroundTaskStatus.isBeginsPatternMatch = true;
+                                    backgroundTaskStatus.isActive = true;
+                                }
+                            }
+                        }
+
+                        if (backgroundTaskStatus.isActive) {
+                            const isEndsPatternMatch = collector.matchEndMatcher(event.line);
+                            // Mark ends pattern as matches, only after begins pattern matches
+                            if (isEndsPatternMatch) {
+                                backgroundTaskStatus.isEndsPatternMatch = true;
+                                this.fireBackgroundTaskEndedEvent({
+                                    taskId: event.taskId,
+                                    ctx: event.ctx
+                                });
```

Comment:
Hi @ShimonBenYair 

quick question:
as you can see from the example in this comment, the beginsPattern - endsPattern could be found multiple times in the task output. So ... how many BackgroundTaskEndedEvent are we going to fire? Do we just fire it on finding the first beginsPattern - endsPattern, or every time the beginsPattern - endsPattern is found ?

Reply:
Hi @elaihau 

When we first find the "beginsPattern - endsPattern " we fire the 'BackgroundTaskEndedEvent '.
And after the first 'BackgroundTaskEndedEvent ' was fired, for every time the endsPattern is found we fire the 'BackgroundTaskEndedEvent ' ( since we already know that the task is active in the background)

Reply:
OK in that case i guess removing `backgroundTaskStatus.isBeginsPatternMatch` and `backgroundTaskStatus.isEndsPatternMatch` wouldn't change the end effect of your function, since you added `isActive`. What do you think? :)

Reply:
fixed, done.